### PR TITLE
S3guard/hadoop 13786 committer

### PR DESCRIPTION
**Pull request by ehiggs**
Various small fixes to the Hadoop s3guard patch set.

### CODE REVIEW

**Comment by ehiggs** [comment]
Rebased.

**Comment by steveloughran** [comment]
heh, I've just gone and broken things by squashing patches

File: hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/commit/magic/MagicS3GuardCommitter.java [hunk 1]
```diff
@@ -22,9 +22,11 @@
 import java.util.Collections;
 import java.util.List;
 
+import org.apache.hadoop.fs.s3a.commit.files.PendingSet;
```
Comment (steveloughran) [inline hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/commit/magic/MagicS3GuardCommitter.java]: don't like imports to be moved around. As well as a nominal rule of layout, changes here make merging a nightmare

File: hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/s3a_committer_architecture.md [hunk 2]
```diff
@@ -951,23 +951,23 @@ to the job attempt.
 1. The task commit operation lists all `.pending` files in its attempt directory.
 1. The contents are loaded into a list of single pending uploads.
 1. These are merged into to a single `Pendingset` structure.
-1. Which is saved to a `.pendingset` file in the job attempt directory.
+1. Which is saved to a `.pendingSet` file in the job attempt directory.
```
Comment (steveloughran) [inline hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/s3a_committer_architecture.md]: no. it's "pendingset" lower case.

File: hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/commit/CommitOperations.java [hunk 3]
```diff
@@ -30,9 +30,11 @@
 import com.amazonaws.services.s3.model.UploadPartRequest;
 import com.amazonaws.services.s3.model.UploadPartResult;
 import com.google.common.base.Preconditions;
+import org.apache.hadoop.fs.s3a.commit.files.PendingSet;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import org.apache.commons.lang3.tuple.Pair;
```
Comment (steveloughran) [inline hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/commit/CommitOperations.java]: never knew of this. good to use something else by other people.

File: hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/commit/files/PendingSet.java [hunk 4]
```diff
@@ -47,8 +47,8 @@
  */
 @InterfaceAudience.Private
 @InterfaceStability.Unstable
-public class Pendingset extends PersistentCommitData {
```
Comment (steveloughran) [inline hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/commit/files/PendingSet.java:50]: Ignoring the fact that case-changing classnames is something very unlikely to take reliably in macos unless you can delete the old one first, I'm unsure about this. Why. It's not really a "set" in terms of java lang collection. At the same time, I'm not that fussy about classnames, though I do want it to stay all lower in the filesystem.

**Comment by steveloughran** [comment]
overall

+1 to the new pair class, guava stripping, etc
=0 to the pendingset/pendingSet classchange; -1 to making the filename mixed case
-1 to moving imports around; especially when it doesn't follow the (nominal) hadoop ordering

File: hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/commit/files/PendingSet.java (hunk 4 above)
Comment (ehiggs) [reply hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/commit/files/PendingSet.java:50]: Well it's two words. Pending and Set. So it's camel cased. Maybe `PendingBatch`?

The issue about case changing and macos doesn't make sense. The patch wasn't merged yet so no one will have issues with the name change.

**Comment by ehiggs** [comment]
I rebased and added a few patched addressing @steveloughran's comments.

I also fixed up some grammar or sentence fragments in the architecture doc.

**Comment by steveloughran** [comment]
thanks, merged in

**Pull request merged**